clang/lib/CIR/CodeGen/CIRGenBuiltinX86.cpp

```diff
@@ -1662,12 +1662,43 @@ CIRGenFunction::emitX86BuiltinExpr(unsigned builtinID, const CallExpr *expr) {
   case X86::BI__builtin_ia32_cmpnltsd:
   case X86::BI__builtin_ia32_cmpnlesd:
   case X86::BI__builtin_ia32_cmpordsd:
+    cgm.errorNYI(expr->getSourceRange(),
+                 std::string("unimplemented X86 builtin call: ") +
+                     getContext().BuiltinInfo.getName(builtinID));
+    return {};
   case X86::BI__builtin_ia32_vcvtph2ps_mask:
   case X86::BI__builtin_ia32_vcvtph2ps256_mask:
   case X86::BI__builtin_ia32_vcvtph2ps512_mask:
   case X86::BI__builtin_ia32_cvtneps2bf16_128_mask:
   case X86::BI__builtin_ia32_cvtneps2bf16_256_mask:
-  case X86::BI__builtin_ia32_cvtneps2bf16_512_mask:
+  case X86::BI__builtin_ia32_cvtneps2bf16_512_mask: {
+    mlir::Location loc = getLoc(expr->getExprLoc());
+    llvm::StringRef intrinsicName;
+    switch (builtinID) {
+    case X86::BI__builtin_ia32_vcvtph2ps_mask:
+      intrinsicName = "x86.avx512.mask.vcvtph2ps.128";
+      break;
+    case X86::BI__builtin_ia32_vcvtph2ps256_mask:
+      intrinsicName = "x86.avx512.mask.vcvtph2ps.256";
+      break;
+    case X86::BI__builtin_ia32_vcvtph2ps512_mask:
+      intrinsicName = "x86.avx512.mask.vcvtph2ps.512";
+      break;
+    case X86::BI__builtin_ia32_cvtneps2bf16_128_mask:
+      intrinsicName = "x86.avx512bf16.mask.cvtneps2bf16.128";
+      break;
+    case X86::BI__builtin_ia32_cvtneps2bf16_256_mask:
+      intrinsicName = "x86.avx512bf16.cvtneps2bf16.256";
+      break;
+    case X86::BI__builtin_ia32_cvtneps2bf16_512_mask:
```

Comment:
You need to follow the intrinsic with a select operation for BI__builtin_ia32_cvtneps2bf16_256_mask and BI__builtin_ia32_cvtneps2bf16_512_mask